File: MetaBrain/providers/groq/adapter.ts
import { buildProviderResponse } from "../provider-response";
import type { ProviderRequest, ProviderResponse } from "../types";
import { GROQ_CAPABILITIES } from "./capabilities";
import { groqHealthDisabled } from "./health";
import type { GroqProviderAdapter } from "./contract";

export class DisabledGroqAdapter implements GroqProviderAdapter {
  readonly provider_name = "groq" as const;
  readonly model_name = "groq-disabled";
  readonly capabilities = GROQ_CAPABILITIES;

  async complete(input: ProviderRequest): Promise<ProviderResponse> {
    return buildProviderResponse({
      request: input,
      provider_name: "groq",
      model_name: this.model_name,
      status: "disabled",
      safety_flags: ["PROVIDER_DISABLED", "NO_EXTERNAL_CALL"],
    });
  }

  async healthcheck() {
    return groqHealthDisabled();
  }
}

File: MetaBrain/providers_py/future_medical/adapter.py
from __future__ import annotations

from MetaBrain.providers_py.provider_health import disabled_provider_health
from MetaBrain.providers_py.provider_response import build_provider_response
from MetaBrain.providers_py.types import ProviderAdapter, ProviderCapabilities, ProviderRequest, ProviderResponse


class DisabledFutureMedicalAdapter(ProviderAdapter):
    def __init__(self) -> None:
        super().__init__(
            provider_name="future-medical",
            model_name="future-medical-disabled",
            capabilities=ProviderCapabilities(True, True, True, True, False, True, 16384, False),
        )

    def complete(self, request: ProviderRequest) -> ProviderResponse:
        return build_provider_response(
            request,
            provider_name="future-medical",
            model_name=self.model_name,
            status="disabled",
            safety_flags=["NO_EXTERNAL_CALL", "NO_MEDICAL_DIAGNOSIS"],
        )

    def healthcheck(self):
        return disabled_provider_health("future-medical")
